hw8/task2.js

```diff
@@ -0,0 +1,314 @@
+        },
+        {
+        id: 7,
+        name: "Отдел маркетинга",
+        employees_count: 20,
+        },
+        {
+        id: 8,
+        name: "Отдел охраны труда",
+        employees_count: 5,
+        },
+    ]
+    },
+    {
+    id: 9,
+    name: "Предприятие 3",
+    departments: [
+        {
+        id: 10,
+        name: "Отдел аналитики",
+        employees_count: 0,
+        },
+    ]
+    }
+]
+
+/*Задания:
+1. Вывести все предприятия и их отделы. Рядом указать количество сотрудников. 
+Для предприятия посчитать сумму всех сотрудников во всех отделах.
+
+**Пример:**
+
+Предприятие 1 (45 сотрудников)
+- Отдел тестирования (10 сотрудников)
+- Отдел маркетинга (20 сотрудников)
+- Администрация (15 человек)
+Предприятие 2 (75 сотрудников)
+- Отдел разработки (50 сотрудников)
+- Отдел маркетинга (20 сотрудников)
+- Отдел охраны труда (5 сотрудников)
+Предприятие 3 (нет сотрудников)
+- Отдел аналитики (нет сотрудников)
+*/
+
+function printCompanyInfo(arr){
+    arr.forEach((enterprise) => {
+        const totalEmployeeCount = enterprise.departments.reduce((sum, department) => sum + department.employees_count, 0);
+        if (totalEmployeeCount > 0) {
+            console.log(`${enterprise.name} (${totalEmployeeCount} сотрудников)`);
+        } else {
+            console.log(`${enterprise.name} (нет сотрудников)`);
+        }
+        enterprise.departments.forEach((department) => {
+        if (department.employees_count > 0) {
+            console.log(`- ${department.name} (${department.employees_count} сотрудников)`)
+        } else {
+            console.log(`- ${department.name} (нет сотрудников)`);
+        };
+        })
+    })
+}
+
+printCompanyInfo(enterprises);
+console.log('=======')
+
+/*
+2. Написать функцию, которая будет принимать 1 аргумент (id отдела или название отдела 
+и возвращать название предприятия, к которому относится).
+
+Пример:
+getEnterpriseName(4) // Предприятие 1
+getEnterpriseName("Отдел маркетинга") // Предприятие 2
+
+function getCompanyName(idOrName) {
+    const company = 
+}
+*/
+
+function getEnterpriseName(nameOrId) {
+    const company = getCompanyByNameOrId(nameOrId);
+    return company.name;
+}
+
+function getCompanyByNameOrId(nameOrId){
+    const foundCompany = enterprises.find(company => {
+        const foundDept = company.departments.find((dept) => dept.name === nameOrId || dept.id === nameOrId);
+        return foundDept;
+    })
+    if (!foundCompany) throw new Error("No such enterprise");
+    return foundCompany;
+}
+
+console.log(getEnterpriseName(7));
+console.log('=======')
+/*
+3. Написать функцию, которая будет добавлять предприятие. В качестве аргумента принимает название предприятия
+
+Пример:
+addEnterprise("Название нового предприятия")
+*/
+
+function addEnterprise(name) {
+    const newEnterprise = {
+        id: getNewId(),
+        name,
+        departments: []
+    }
+    enterprises.push(newEnterprise)
+}
+
+function getNewId(id) {
+    id = 0;
```

Comment:
```suggestion
function getNewId() {
    let id = 0;
```
You don't need this argument